### PR TITLE
GitHub Proxy part 1: github integration resource

### PR DESCRIPTION
**Pull request by greedy52**
Part of:
- #48762

Changes:
- New github integration spec added to `integration` resource
- Added `WithSecrets` to `GetIntegration` and `ListIntegrations`
- Auth to populate per-app CA when creating github integration

Many files were touched but most are one-liner for the interface change.

`tctl` create example:
```yaml
kind: integration
sub_kind: github
version: v1
metadata:
  name: github-my-org
spec:
  github:
    organization: my-org
  credentials:
    # oauth id and secret
    id_secret:
      id: "my-id"
      secret: "my-secret"
```

### CODE REVIEW

File: api/types/integration.go [hunk 1]
```diff
@@ -230,6 +267,49 @@ func (s *IntegrationSpecV1_AzureOIDC) Validate() error {
 	return nil
 }
 
+// CheckAndSetDefaults validates the configuration for GitHub integration subkind.
+func (s *IntegrationSpecV1_GitHub) CheckAndSetDefaults() error {
+	if s == nil || s.GitHub == nil {
+		return trace.BadParameter("github is required for %q subkind", IntegrationSubKindGitHub)
+	}
+	if s.GitHub.Organization == "" {
```
Comment (r0mant) [inline api/types/integration.go:275]: Can we validate organization name?

**Review by r0mant — APPROVED** [review]

**Review by smallinsky — APPROVED** [review]

**Comment by r0mant** [comment]
@greedy52 Let's use plugin_static_credentials. They are specifically for storing secrets for hosted plugins and integrations.

File: api/types/integration.go (hunk 1 above)
Comment (greedy52) [reply api/types/integration.go]: added some validation in `api/types/integration_github.go`

**Comment by greedy52** [comment]
> @greedy52 Let's use plugin_static_credentials. They are specifically for storing secrets for hosted plugins and integrations.

Moved to static credentials now. PTAL.

**Review by r0mant — APPROVED** [review]

**Review by GavinFrazar — APPROVED** [review]

File: api/proto/teleport/legacy/types/types.proto [hunk 2]
```diff
@@ -7015,6 +7016,14 @@ message PluginStaticCredentialsOAuthClientSecret {
   string ClientSecret = 2 [(gogoproto.jsontag) = "client_secret"];
 }
 
+// PluginStaticCredentialsSSHCertAuthorities contains the active SSH CAs used
+// for the integration or plugin.
+message PluginStaticCredentialsSSHCertAuthorities {
+  // CertAuthorities contains the active SSH CAs used for the integration or
+  // plugin.
+  repeated SSHKeyPair cert_authorities = 1;
```
[the remark this answers is not in the document]
Comment (greedy52) [reply api/proto/teleport/legacy/types/types.proto:7149]: the default one is good:
```
type PluginStaticCredentialsSSHCertAuthorities struct {                                                     
    // CertAuthorities contains the active SSH CAs used for the integration or                  
    // plugin.                                                                                         
    CertAuthorities      []*SSHKeyPair `protobuf:"bytes,1,rep,name=cert_authorities,json=certAuthorities,proto3" json:"cert_authorities,omitempty"`
```

File: api/types/integration.go [hunk 3]
```diff
@@ -333,6 +401,13 @@ func (ig *IntegrationV1) UnmarshalJSON(data []byte) error {
 	}
 
 	integration.ResourceHeader = d.ResourceHeader
+	if len(d.Spec.Credentials) != 0 {
+		var credentials PluginCredentialsV1
+		if err := protojson.Unmarshal(d.Spec.Credentials, protoadapt.MessageV2Of(&credentials)); err != nil {
```
[the remark this answers is not in the document]
Comment (greedy52) [reply api/types/integration.go:406]: it does. we've been using it in services:
https://github.com/gravitational/teleport/blob/e8b3d217d2965dfef4fa0171dab014e362bfc104/lib/services/plugin_static_credentials.go#L86-L87

**Comment by greedy52** [comment]
Thanks for the quick reviews!

**Pull request merged**